### PR TITLE
Feature/improve infrastructure readme

### PR DESCRIPTION
**Pull request by Franco0712**
## Description:
Change README adding new documentation for deploy and a new table of content for extra infromation and configuration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,117 +7,41 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
+1. Build deployment image: `docker build -t video-headline-deploy .`
 
-Create a .env file at the root of the project with all the variables defined in the .env-example file and their respective values.
+2. Run deployment: `docker run -e AWS_ACCESS_KEY_ID=... -e AWS_SECRET_ACCESS_KEY=... -e AWS_SESSION_TOKEN=... -e PROCESS=deploy -it video-headline-deploy`
 
-#### AWS Configuration
-Video Headline requires some IAM roles and permissiones. To automate the configuration, there's CDK code to create a Stack with all the requirements.
+   - AWS_ACCESS_KEY_ID: AWS access key identifier.
+   - AWS_SECRET_ACCESS_KEY: AWS secret access key.
+   - AWS_SESSION_TOKEN: AWS session token (if required).
```
Comment (JoselenCeciliaQ) [inline README.md]: add where find the keys

File: infrastructure/README.md [hunk 2]
```diff
@@ -1,7 +1,38 @@
...
+### Prerequisites
+- AWS account (it's not necessary to have any profile configured locally).
+- Docker running on your machine.
+### Steps to deploy Videoheadline Infrastructure
+1. Build deployment image: `docker build -t video-headline-deploy .`
+
+2. Run deployment: `docker run -e AWS_ACCESS_KEY_ID=... -e AWS_SECRET_ACCESS_KEY=... -e AWS_SESSION_TOKEN=... -e PROCESS=deploy -it video-headline-deploy`
+    - AWS_ACCESS_KEY_ID: AWS access key identifier.
+    - AWS_SECRET_ACCESS_KEY: AWS secret access key.
+    - AWS_SESSION_TOKEN: AWS session token (if required).
```
Comment (JoselenCeciliaQ) [inline infrastructure/README.md:25]: add where find the keys

File: README.md [hunk 3]
```diff
@@ -7,114 +7,43 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
-    - [Create .env file](#create-.env-file)
-    - [AWS Configuration](#aws-configuration)
-  - [Running the application in local environment](#running-the-application-in-local-environment)
-    - [Set up and running the application](#set-up-and-running-the-application)
-    - [AWS Services Configuration in the Admin Web](#aws-services-configurationiin-the-admin-web)
-  - [Running the application in AWS environment](#running-the-application-in-aws-environment)
-  - [Production Environment Tasks (optional)](#production-environment-tasks)
-  - [Custom CSS Configuration for the Player in an Organization](#Custom-css-configuration-for-the-player-in-an-organization)
+Deploy a stable version of VideoHeadline using a pre-built docker-hub image for a quick and easy deployment process.
+Alternatively, you can build the image locally to deploy a customized version of VideoHeadline, changing the code it's configuration files (see [Table of extra contents](#table-of-extra-contents) for more information).
```
Comment (sgrazi) [inline README.md]:
```suggestion
Alternatively, you can build the image locally to deploy a customized version of VideoHeadline, changing the code and it's configuration files (see [Table of extra contents](#table-of-extra-contents) for more information).
```

File: README.md [hunk 4]
```diff
@@ -7,114 +7,43 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
-- Yarn and Node.js(v10): Required to deploy AWS configurations and build the playerReact component.
-- Python: Necessary for running Django and other Python-based tools.
-- AWS CLI: Useful for configuring and managing AWS services from the command line.
+- AWS account (it's not necessary to have any profile configured locally).
+- Docker running on your machine.
 
-#### Create .env file
+### Steps to deploy Videoheadline Infrastructure
 
-Create a .env file at the root of the project with all the variables defined in the .env-example file and their respective values.
```
Comment (sgrazi) [inline README.md:36]: why was this deleted, is it no longer necessary to have a .env?

**Review by sgrazi — APPROVED** [review]
small question, overall looks good

File: README.md [hunk 5]
```diff
@@ -7,114 +7,43 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
-    - [Create .env file](#create-.env-file)
-    - [AWS Configuration](#aws-configuration)
-  - [Running the application in local environment](#running-the-application-in-local-environment)
-    - [Set up and running the application](#set-up-and-running-the-application)
-    - [AWS Services Configuration in the Admin Web](#aws-services-configurationiin-the-admin-web)
-  - [Running the application in AWS environment](#running-the-application-in-aws-environment)
-  - [Production Environment Tasks (optional)](#production-environment-tasks)
-  - [Custom CSS Configuration for the Player in an Organization](#Custom-css-configuration-for-the-player-in-an-organization)
+Deploy a stable version of VideoHeadline using a pre-built docker-hub image for a quick and easy deployment process.
+Alternatively, you can build the image locally to deploy a customized version of VideoHeadline, changing the code and it's configuration files (see [Table of extra contents](#table-of-extra-contents) for more information).
```
Comment (NoeliaBentancor) [inline README.md]:
```suggestion
Alternatively, you can build the image locally to deploy a customized version of VideoHeadline, changing the code and its configuration files (see [Table of extra contents](#table-of-extra-contents) for more information).
```

File: README.md [hunk 6]
```diff
@@ -7,114 +7,43 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
-- AWS Account: Necessary for hosting and delivering video content.
-- Docker and Docker compose: Video Headline runs inside Docker containers, so it is necessary to have Docker and Docker Compose installed.
-- Yarn and Node.js(v10): Required to deploy AWS configurations and build the playerReact component.
-- Python: Necessary for running Django and other Python-based tools.
-- AWS CLI: Useful for configuring and managing AWS services from the command line.
+- AWS account (it's not necessary to have any profile configured locally).
+- Docker running on your machine.
 
-#### Create .env file
+### Steps to deploy Videoheadline Infrastructure
```
Comment (NoeliaBentancor) [inline README.md]:
```suggestion
### Steps to deploy VideoHeadline Infrastructure
```

File: README.md [hunk 7]
```diff
@@ -7,114 +7,43 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
 
-To deploy this stack, follow this steps:
-1. Navigate to the infrastructure directory.
-2. Run the command: `yarn cdk deploy AwsConfigurationStack`.
+   - AWS_ACCESS_KEY_ID: AWS access key identifier.
+   - AWS_SECRET_ACCESS_KEY: AWS secret access key.
+   - AWS_SESSION_TOKEN: AWS session token (if required).
 
-This deployment will set up:
+   These variables can be found in AWS Command line or programmatic access.
```
Comment (NoeliaBentancor) [inline README.md:30]: Is this necessary?

File: README.md [hunk 8]
```diff
@@ -7,114 +7,43 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
-  - MediaLive
-  - Cloudfront
-  - Cloudwatch
-- Media Convert Role with permissions for:
-  - Api Gateway
-  - S3
-- Media Live Role with permissions for:
-  - MediaLive
-  - Cloudwatch
+3. Once the implementation process has started through the console, you may be asked to confirm with a y/n, please confirm it.
```
Comment (NoeliaBentancor) [inline README.md:32]: I will summary this step

File: infrastructure/README.md [hunk 9]
```diff
@@ -1,7 +1,41 @@
 # Video Headline CDK Deployment
-The `cdk.json` file tells the CDK Toolkit how to execute your app.
+# Table of Contents
+- Deployment
```
Comment (sgrazi) [inline infrastructure/README.md]:
```suggestion
- [Deployment](#Deployment)
```

File: infrastructure/README.md [hunk 10]
```diff
@@ -1,7 +1,41 @@
...
+    - [Use pre-build image](#Use-pre-build-image)
+    - [Use local image](#use-local-image)
+- Destroy
```
Comment (sgrazi) [inline infrastructure/README.md]:
```suggestion
- [Destroy](#Destroy)
```

File: README.md [hunk 11]
```diff
@@ -7,114 +7,43 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
+4. The url of the application will be displayed in the console
 
-**Environment Variables:** Add `AWS_MEDIA_CONVERT_ROLE` and `AWS_MEDIA_LIVE_ROLE` with respective ARNs to your Docker Compose file based on your environment (`docker-compose.dev.yml` or `docker-compose.prod.yml`).
+5. At the end of this deploy process, a superuser will be created to use in the application, you will be asked via console for the data you want to use.
 
-#### Set up and running the application
+### Accessing the App
 
-Follow these steps to set up and run the application locally:
+Once the app is deployed you can access the web through the previously mentioned url. If you want to access the Admin web you need to add /admin to the Distribution domain name.
```
Comment (NoeliaBentancor) [inline README.md]: I will list both or refer them

File: README.md [hunk 12]
```diff
@@ -7,114 +7,43 @@ system (CMS). It is a deployable solution that leverages the power of AWS servic
...
-
-1. Have a URL ready with a CSS file to test (not a local path).
-2. In the Organization, add the following line in the organization configuration field:
-
-```
-  "playerCustomCss":"https://your-css-url/cssFile.css"
-```
-
-If you do not have a CSS URL, you can serve a folder using the npx serve command: 1. Navigate to the folder containing the CSS file you want to use. 2. Run the command 'npx serve .'. 3. Open the server from the browser (probably http://localhost:5000), find the file, and copy the URL.
+- Deploy videoheadline to AWS and
```
Comment (NoeliaBentancor) [inline README.md]:
```suggestion
- Deploy VideoHeadline to AWS and
```

File: infrastructure/README.md [hunk 13]
```diff
@@ -1,7 +1,41 @@
 # Video Headline CDK Deployment
```
Comment (NoeliaBentancor) [inline infrastructure/README.md]:
```suggestion
# VideoHeadline CDK Deployment
```

File: infrastructure/README.md [hunk 14]
```diff
@@ -1,7 +1,41 @@
...
+    - [Use local image](#use-local-image)
+- Destroy
+    - [Destroy deployment using pre-built image](#destroy-deployment-using-pre-built-image)
+    - [Destroy deployment using local image](#destroy-deployment-using-local-image)
+- [Useful commands](#useful-commands)
+- [Configuring HTTPS](#configuring-https)
 
-## Getting started
+# Deployment
+To deploy videoheadline to AWS there are two alternatives. The first involves using a stable videoheadline version (main), so there is already a built image that exists in docker-hub and the deploy process will use this image. The second alternative involves building the videoheadline image before performing the deploy process, this can be used to make changes locally and to perform a deploy without depending on the version of main
```
Comment (NoeliaBentancor) [inline infrastructure/README.md]:
```suggestion
To deploy VideoHeadline in AWS there are two alternatives. The first involves using a stable VideoHeadline version (main), so there is already a built image that exists in docker-hub and the deploy process will use this image. 
The second alternative involves building VideoHeadline image before performing  deploy process, this can be used to make changes locally and to perform a deploy without depending on the version of main
```

File: infrastructure/README.md [hunk 15]
```diff
@@ -1,7 +1,41 @@
...
+## Use pre-build image
+### Prerequisites
+- AWS account (it's not necessary to have any profile configured locally).
+- Docker running on your machine.
+### Steps to deploy Videoheadline Infrastructure
```
Comment (NoeliaBentancor) [inline infrastructure/README.md]:
```suggestion
### Steps to deploy VideoHeadline Infrastructure
```

File: infrastructure/README.md [hunk 16]
```diff
@@ -10,36 +44,52 @@ The `cdk.json` file tells the CDK Toolkit how to execute your app.
...
-### Accessing the App
-Once the app is deployed you can access the web through Videoheadline Cloudfront using the Distribution domain name.
-If you want to access the Admin web you need to add /admin to the Distribution domain name.
-# Creating superuser
-## Prerequisites
+5. Once the implementation process has started through the console, you may be asked to confirm with a y/n, please confirm it.
+
+6. The url of the application will be displayed in the console
+
+### Creating superuser
```
Comment (NoeliaBentancor) [inline infrastructure/README.md:58]: What goes here?

File: infrastructure/README.md [hunk 17]
```diff
@@ -10,36 +44,52 @@ The `cdk.json` file tells the CDK Toolkit how to execute your app.
...
+#### Prerequisites
 - You need to install AWS Session manager plugin (https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html).
 - You need to configure your AWS_PROFILE env variable.
 
-## Running the create_super_user.py script
+#### Running the create_super_user.py script
```
Comment (NoeliaBentancor) [inline infrastructure/README.md]:
```suggestion
#### Running  create_super_user.py script
```

File: infrastructure/README.md [hunk 18]
```diff
@@ -10,36 +44,52 @@ The `cdk.json` file tells the CDK Toolkit how to execute your app.
...
 To create a superuser run `python create_super_user.py`
```
Comment (NoeliaBentancor) [inline infrastructure/README.md]:
```suggestion
Run `python create_super_user.py`
```

File: infrastructure/README.md [hunk 19]
```diff
@@ -10,36 +44,52 @@ The `cdk.json` file tells the CDK Toolkit how to execute your app.
...
 -   `yarn cdk diff` compare deployed stack with current state
 -   `yarn cdk synth` emits the synthesized CloudFormation template
-## Configuring HTTPS
-If you want to configure https with a certificate you can put your certificate arn in `/lib/utils/aws.ts`. (https://github.com/qualabs/video-headline/blob/cca91831ef0130ea1fefa53084b949308d55c57e/infrastructure/lib/utils/aws.ts#L25).
-
-# Remove
-
-If you want to remove all the resources deployed you can run `yarn cdk destroy '*' --profile [...]`
+# Configuring HTTPS
+If you want to configure https with a certificate you can put your certificate arn in `/lib/utils/aws.ts`. (https://github.com/qualabs/video-headline/blob/cca91831ef0130ea1fefa53084b949308d55c57e/infrastructure/lib/utils/aws.ts#L25).
```
Comment (NoeliaBentancor) [inline infrastructure/README.md]: What do you mean by "putting your certificate arn"?

File: README.md (hunk 7 above)
Comment (Franco0712) [reply README.md:30]: I think its necessary in case the user does not know where the variables are

File: README.md (hunk 11 above)
Comment (Franco0712) [reply README.md]: The URL is logged in the command line

File: infrastructure/README.md (hunk 19 above)
Comment (Franco0712) [reply infrastructure/README.md]: It was not necessary

Comment (Franco0712) [reply infrastructure/README.md]: It was not necessary

**Review by lucilamoreno — APPROVED** [review]

**Review by NoeliaBentancor — APPROVED** [review]
